### ISSUE TITLE
FP2: battery consumption much too high when mobile data is switched on

### ISSUE THREAD

**Issue by derkleinemannvomfeldweg** [ubports/ubuntu-touch#400]
- Device: FP2
- Channel: OAT 3 / stable
- Build:

### Steps to reproduce
Switch on mobile data

### Expected behavoir
Provide a battery life comparable to Android (Lineage OS: several days when just lying around)

### Actual behavior
Instead, batterydrains in 12... 14 hours. Only using wifi, battery life is as expected.
Behavior is independend on the chosen type of connection(2G/3G/4G).

### Logfiles and additional information 
If logfiles are needed, I can provide them. I could also provide screenshots with battery graph from the settings, if needed.

**Comment by kwah-gh** [comment]
I actually observe the same issue, when only Wi-Fi is enabled. This is one of the issues stopping me from using my FP2 as daily device, i.e. there is no SIM-card/mobile connection on my FP2 yet.

**Comment by derkleinemannvomfeldweg** [comment]
Is it maybe a duplicate of #387 ?

**Comment by NeoTheThird** [comment]
Since it's two different devices and the problem is not that present on other devices, let's track it in separate issues.

**Comment by delijati** [comment]
Same for BQ Aquaris E 4.5 if i deactivate mobile data its lasts 4 days as usual .. some service polling to much ??

**Comment by DiogoConstantino** [comment]
I confirm this issue, it also happens with me.

**Comment by mgclude** [comment]
I have both the FP2 and BQ e4.5 and I have the same issue on both of them. My BQ e4.5 worked fine (battery lasting 2 days on average usage) with canonical image. Since switching to Ubports image, battery drops quick, lasts not even a full day, with same or less usage than before. The battery graph drains steadily, even on idle (no usage, mobile networks on).

**Comment by DiogoConstantino** [comment]
I also see this happening with other devices even when using only wifi (bq 4.5 and pro5), or even in airplane mode (seen only on the pro5).

**Comment by Flohack74** [comment]
Guys can you please open issue #415 and follow the config change that was made at the end to the push client config, and re-test everything? Thx!

**Comment by derkleinemannvomfeldweg** [comment]
I changed the config like suggested, and now the battery consumption is like I would expect it.

**Comment by Flohack74** [comment]
Ok thats good to hear. Do you face any shortcomings now, like long delay of notifications etc?

**Comment by derkleinemannvomfeldweg** [comment]
To be honest, I can't tell you exactly, as I never tried sending myself telegram messages or emails. I will check tomorrow, when I have another device at hand.

**Comment by derkleinemannvomfeldweg** [comment]
I get telegram notifications without a significant delay, as far as I can tell. Dekko2 notifies me with a delay of 30 minutes or so, but manually checking the inboxes revealed that the messages haven't been fetched from the server already, so in fact there was nothing to notify.

**Comment by jezek** [comment]
I have a battery drain problem on my FP2 too. If wifi is on (with mobile data off, bluetooth off and ubuntu-push-client stopped), the phone lasts only about 24h. When I turn off the wifi, the phone lasts as expected (idle time is like in energy use in [fp2 specs](https://support.fairphone.com/hc/en-us/articles/203478389-What-are-the-technical-specifications-of-the-Fairphone-2-)). So my bet on battery drain is wifi.

PS: I have a personal log on phone battery status & usage for some statistics if needed. The concluson above is based on them (more trials with different/same configurations)
PPS: I did not install firmware upgrade yet, to collect battery drain data before and after.

**Comment by kwah-gh** [comment]
Miserable battery life still makes FP2 unusable for me. Meanwhile I've acquired one more FP2 (looks like the same build model and HW components, although I did not compare them thoroughly).

In the same usage mode as FP2 with UBports (WiFi only, no SIM card, basically idle all the time), FP2 with Android drained about 50% of battery in about 5 days.

I could help with experiments / collecting logs / data etc. Any ideas where one should start, while trying to figure out what is going on?

**Comment by Flaburgan** [comment]
Yes I agree the issue title should be edited, the problem is not mobile data, it is Wi-Fi connection.

**Comment by jezek** [comment]
I've made an test, with wifi on, mobile data on.
![screenshot20210208_183010454](https://user-images.githubusercontent.com/5408235/107393458-1feea080-6afb-11eb-85d3-520820048944.png)
Note: The 2 steps are browsing time.
Note: The FP2 was bought on April 2018 as new.

**Comment by capsia37** [comment]
Hello, following the last post on this issue a PR was opened at the devices website to change the "Battery lifetime" feature to the working status. Can anyone confirm this is true for both the 5.1 based port and the Halium 7.1 port? Thanks

**Comment by jezek** [comment]
@capsia37 I confirm for Halium 5.1 port. See the battery graph above.

**Comment by Ublublu** [comment]
Hello, I disagree, with mobile data on, my battery drop very fast and does not last 24h. 
this is with OTA-19, OTA-18.  Battery life time seemed better on OTA-17.

**Comment by mgclude** [comment]
Same happens on wifi. In fact, battery goes down steadily duting the day,
even with data turned off (mobile or wifi). I tested this on several
occasions, with the phone connected to the 2G mobile network, and with data
disabled. Surprisingly, even with the phone staying all day on the exact
same location in the house (no cell tower hopping) the battery is still
drained consistently.
I even turned off notifications with the command "stop ubuntu-push-client"
on terminal: same result. Battery graph goes down during the day until it
runs out, even at idle. No calls, no browsing, no open applications,
nothing. This behaviour is consistent over several days and weeks.

**Comment by mgclude** [comment]
Adding that I replicated the same scenario on my BQe4.5 running the same
version of Ubuntu and this doesn't happen. When idle, battery preserves
charge, with only 1% or 2% drop over 24 hours.

Le jeu. 30 sept. 2021 à 12:38, Miguel Gomes da Costa ***@***.***> a
écrit :

**Comment by Flaburgan** [comment]
@mgclude can you please check with htop with process is using the CPU?

Personally with my FP2 I had that problem only with Wi-Fi. But now I flashed  /e/ OS on that FP2 and I would still say that enabling Wi-Fi is consuming too much power. I just probably update the firmware.

**Comment by ska67** [comment]
FP2 with OTA-20 on stable channel. Since OTA-18, the battery life is far less than 24 hours.

If I run no apps, no Wifi, only Bluetooth and mobile data (2G/3G/4G), the battery lasts maybe 14h

I bought a new battery a few weeks ago thinking it was a hardware defect, but I was wrong.

**Comment by jezek** [comment]
@ska67 I have a FP2with Ubuntu Touch (halium-5.1), currently on RC channel. Around midnight, I charged battery to 100%, turned cellular/mobile data ON, wifi and bluetooth OFF. After cca 19 hours, the battery is on 74% capacity, that is around 30% points in 24 hours. Which is consistent with my previous battery tests.

Note: Now I realized, you write about bluetooth ON. Maybe the bluetooth is problematic. when you turn bluetooth OFF, will the battery lifetime change?

Note 2: Somwhere I read, that bad mobile/bluetooth/wifi signal can cause bigger battery drain. Just for information.

**Comment by ska67** [comment]
After switching the mobile data transmission manually from 2G/3G/4G to only 2G, the battery now lasts up to 48h again.

Either the 4G modem is broken or a new 4G cell tower was installed nearby in the last few weeks, so my FP2 is now constantly trying to connect to 4G.

@jezek  Thanks for the advice, I will try turning off Bluetooth in the next few days to see if the battery life changes.
